src/actions/configuration/first-time-configuration-action.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+namespace Yoast\WP\SEO\Actions\Configuration;
+
+use Yoast\WP\SEO\Helpers\Options_Helper;
+use Yoast\WP\SEO\Integrations\Admin\Social_Profiles_Helper;
+
+// phpcs:disable Yoast.NamingConventions.ObjectNameDepth.MaxExceeded -- First time configuration simply has a lot of words.
+/**
+ * Class First_Time_Configuration_Action.
+ */
+class First_Time_Configuration_Action {
+
+	/**
+	 * The fields for the site representation payload.
+	 */
+	const SITE_REPRESENTATION_FIELDS = [
+		'company_or_person',
+		'company_name',
+		'company_logo',
+		'company_logo_id',
+		'person_logo',
+		'person_logo_id',
+		'company_or_person_user_id',
+		'description',
+	];
+
+	/**
+	 * The fields for the social profiles payload.
+	 */
+	const SOCIAL_PROFILES_FIELDS = [
+		'facebook_site',
+		'twitter_site',
+		'other_social_urls',
+	];
+
+	/**
+	 * The Options_Helper instance.
+	 *
+	 * @var Options_Helper
+	 */
+	protected $options_helper;
+
+	/**
+	 * The Social_Profiles_Helper instance.
+	 *
+	 * @var Social_Profiles_Helper
+	 */
+	protected $social_profiles_helper;
+
+	/**
+	 * First_Time_Configuration_Action constructor.
+	 *
+	 * @param Options_Helper         $options_helper         The WPSEO options helper.
+	 * @param Social_Profiles_Helper $social_profiles_helper The social profiles helper.
+	 */
+	public function __construct( Options_Helper $options_helper, Social_Profiles_Helper $social_profiles_helper ) {
+		$this->options_helper         = $options_helper;
+		$this->social_profiles_helper = $social_profiles_helper;
+	}
+
+	/**
+	 * Stores the values for the site representation.
+	 *
+	 * @param array $params The values to store.
+	 *
+	 * @return object The response object.
+	 */
+	public function set_site_representation( $params ) {
+		$failures = [];
+
+		foreach ( self::SITE_REPRESENTATION_FIELDS as $field_name ) {
+			if ( isset( $params[ $field_name ] ) ) {
+				if ( $field_name === 'description' && \current_user_can( 'manage_options' ) ) {
+					$result = \update_option( 'blogdescription', $params['description'] );
+					if ( ! $result && $params['description'] === \get_option( 'blogdescription' ) ) {
+						$result = true;
+					}
+				}
+				else {
+					$result = $this->options_helper->set( $field_name, $params[ $field_name ] );
+				}
+				if ( ! $result ) {
+					$failures[] = $field_name;
+				}
+			}
+		}
+
+		if ( \count( $failures ) === 0 ) {
+			return (object) [
+				'success' => true,
+				'status'  => 200,
+			];
+		}
+		return (object) [
+			'success'  => false,
+			'status'   => 500,
+			'error'    => 'Could not save some options in the database',
+			'failures' => $failures,
+		];
+	}
+
+	/**
+	 * Stores the values for the social profiles.
+	 *
+	 * @param array $params The values to store.
+	 *
+	 * @return object The response object.
+	 */
+	public function set_social_profiles( $params ) {
+		$failures = [];
+
+		foreach ( self::SOCIAL_PROFILES_FIELDS as $field_name ) {
+			if ( isset( $params[ $field_name ] ) ) {
```

Comment:
This should have the `@pls78 This should just use the `$failures = $this->social_profiles_helper->set_organization_social_profiles( $params );` line and also the rest of the function should be the same with the one from `Configuration_Workout_Action` right @pls78 ?

Reply:
Right!